mailbox/api/src/main/java/org/apache/james/mailbox/events/MailboxListener.java

```diff
@@ -58,7 +58,7 @@
 public interface MailboxListener {
 
     class MaxQueueSize {
-        public static final MaxQueueSize EMPTY = new MaxQueueSize(0);
+        public static final MaxQueueSize NONE = new MaxQueueSize(0);
 
         public static MaxQueueSize of(int size) {
             Preconditions.checkArgument(size > 0, "MaxQueueSize should be strictly positive, '{}' given", size);
```

Comment:
`new MaxQueueSize(0)` above... Won't work.

Can we consider rather using `Optional.empty` rather than a meaningless invalid object?

Reply:
You'd have to inject it at some point, I do not know how to inject an `Optional`

Reply:
Can't then just simply accept 0 (as no max queue size) and specify it in the doc ?

Reply:
It is the RQ behavior but it makes no sense to specify '0' as infinity, it's a magic configuration value and it should be avoided.